cmd/kubeadm/app/phases/uploadconfig/uploadconfig_test.go

```diff
@@ -19,37 +19,79 @@ package uploadconfig
 import (
 	"testing"
 
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	clientsetfake "k8s.io/client-go/kubernetes/fake"
+	core "k8s.io/client-go/testing"
 	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
 )
 
 func TestUploadConfiguration(t *testing.T) {
 	tests := []struct {
-		name    string
-		cfg     *kubeadmapi.MasterConfiguration
-		wantErr bool
+		name           string
+		createErr      error
```

Comment:
Tiny nitpick: with the naming scheme, it wasn't obvious why `createErr` and `updateErr` (which trigger errors) were separate/different from `wantErr`, which indicates whether or not an error is expected. Maybe `errOnCreate` and `errOnUpdate` would make their intent more obvious.